### ISSUE TITLE
Permissive package.json values are causing backward compatibility problems

### ISSUE THREAD

**Issue by jdmarshall** [babel/babel#11300]
## Bug Report

- [ ] I would like to work on a fix!

**Current Behavior**

babel/cli aggressively updates its dependencies, causing our minifier step to fail:

20-Mar-2020 14:18:08 	Warning: Unknown helper createSuper Use --force to continue.

**Input Code**
- REPL or Repo link if applicable:

```
"dependencies": {
    "@babel/code-frame": "^7.0.0",
    "@babel/generator": "^7.2.2",
    "@babel/helpers": "^7.2.0",
    "@babel/parser": "^7.2.2",
    "@babel/template": "^7.2.2",
    "@babel/traverse": "^7.2.2",
    "@babel/types": "^7.2.2",
```

Using caret means that you have to maintain deep backward compatibility between versions. It would be easier for users if you were using tighter matching, like ~7.2.0 and updating on minor releases.

**Expected behavior/code**

Issues like #11292 would not happen if semver was being used more conservatively across the babel project.

**Babel Configuration (babel.config.js, .babelrc, package.json#babel, cli command, .eslintrc)**
- Filename: `babel.config.js`

```js
{
  "your": { "config": "here" }
}
```

**Environment**

```

```
- Babel version(s): [e.g. v6.0.0, v7.0.0-beta.34]
- Node/npm version: [e.g. Node 8/npm 5]
- OS: [e.g. OSX 10.13.4, Windows 10]
- Monorepo: [e.g. yes/no/Lerna]
- How you are using Babel: [e.g. `cli`, `register`, `loader`]

**Possible Solution**

**Additional context/Screenshots**
Add any other context about the problem here. If applicable, add screenshots to help explain.

**Comment by nicolo-ribaudo** [comment]
We do maintain it. It was a bug introduced in 7.8.0, fixed in 7.8.6 when we discovered it.

If you are still using a version <7.8.6, please upgrade to at least 7.8.7 to receive the bug fix.

**Comment by nicolo-ribaudo** [comment]
The fix is at https://github.com/babel/babel/pull/11298
